website/docs/cli/commands/test.mdx

```diff
@@ -132,3 +134,137 @@ The testing directory must be beneath the main configuration directory, but it c
+{"@level":"info","@message":"example_2.tftest.hcl... fail","@module":"terraform.ui","@testfile":"example_2.tftest.hcl","@timestamp":"2025-01-06T12:19:02.231557Z","test_file":{"path":"example_2.tftest.hcl","progress":"complete","status":"fail"},"type":"test_file"}
+{"@level":"info","@message":"Failure! 1 passed, 1 failed.","@module":"terraform.ui","@timestamp":"2025-01-06T12:19:02.231581Z","test_summary":{"status":"fail","passed":1,"failed":1,"errored":0,"skipped":0},"type":"test_summary"}
+```
+
+### Test output in JUnit XML format, saved to file
+
+Below is the output of the `terraform test -junit-xml=./output.xml` command using the same example files.
+The test output is:
+
+* printed to the terminal in the default, human-readable format, as already described above
+* also saved in JUnit XML format in the specified file
+
+Below is the contents of the file `output.xml`:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?><testsuites>
+  <testsuite name="example_1.tftest.hcl" tests="1" skipped="0" failures="0" errors="0">
+    <testcase name="true_is_true" classname="example_1.tftest.hcl" time="0.001064666"></testcase>
+  </testsuite>
+  <testsuite name="example_2.tftest.hcl" tests="1" skipped="0" failures="1" errors="0">
+    <testcase name="false_is_false" classname="example_2.tftest.hcl" time="0.000870083">
+      <failure message="Test run failed"></failure>
+      <system-err><![CDATA[
+Error: Test assertion failed
+
+  on example_2.tftest.hcl line 21:
+  (source code not available)
+
+local.false did not match expected value
+]]></system-err>
+    </testcase>
+  </testsuite>
+</testsuites>
+```
+
+The file maps Terraform test command concepts to JUnit XML format according to the table below:
+
+| Terraform test concept | Element in JUnit XML output | Notes        |
+| -----------------------| ----------------------------|--------------|
+| Test directory         | `<testsuites>`              | A single `<testsuites>` element wraps all other elements in the XML output, as `terraform test` only processes a single test directory. |
+| Test file              | `<testsuite>`               | There will be as many `<testsuite>` elements as there are test files. The tests in a `<testsuite>` are summarized in its attributes: `name` is the file name; `tests` is total count of tests; `skipped` is a count of skipped tests; `failures` is a count of failed tests, `errors` is a count of errored tests. |
+| Run block              | `<testcase>`                | Any `<testcase>` elements will be nested inside a parental `<testsuite>` element corresponding to the file that contains the run block. |
+| Test failure error     | `<failure>`                 | If a `<testcase>` fails then it will contain a `<failure>` element that contains a message about a test failure. |
+| ???                    | `<system-err>`              | ??? |
```

Comment:
To do!

This will hopefully be made clear during work on a related PR that refactors and adds test coverage to the feature